packages/ec-core/src/operator/selector/tournament.rs

```diff
@@ -1,18 +1,31 @@
+use std::num::NonZeroUsize;
+
 use anyhow::{ensure, Context, Result};
 use rand::{prelude::IndexedRandom, rngs::ThreadRng};
 
 use super::Selector;
 use crate::population::Population;
 
 pub struct Tournament {
-    size: usize,
+    size: NonZeroUsize,
 }
 
 impl Tournament {
+    /// Construct a tournament selector with the given size. This
+    /// will select `size` individuals from the population, randomly
+    /// without replacement, and return the "best" from that set.
     #[must_use]
-    pub const fn new(size: usize) -> Self {
+    pub const fn new(size: NonZeroUsize) -> Self {
         Self { size }
     }
+
+    /// Construct a binary tournament selector, i.e., a tournament
+    /// selector that selects two random individuals from the population
+    /// and returns the "better" of the two.
+    #[must_use]
+    pub const fn binary() -> Self {
+        Self::new(NonZeroUsize::MIN.saturating_add(1))
```

Comment:
This rather odd way of creating `2` allows `binary()` to be `const`, and avoids any chance of failure and need for `unwrap()` calls. That then avoids the need for all the complex `allow/expect` stuff here.